### PR TITLE
feat(panel): Add slot for an action-bar component.

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #6448

## Summary

Panel gains an "action bar slot" beneath the rendered header to provide familiar and consistent placement of actions in a workflow setting.

### CODE REVIEW

**Comment by asangma** [comment]
@driskull One note. Does content in the action-bar slot stay sticky along with the header node? Seems like we would want that behavior.

**Comment by driskull** [comment]
@asangma no, that was not defined in the requirements. It would need to be in the header then and not below.

**Comment by macandcheese** [comment]
I think it's working as expected currently: 


https://user-images.githubusercontent.com/4733155/230679518-03a73fda-fd2c-4c0e-9eab-5f5c9e00da16.mov

The only thing that sticks out is part of Action Bar - the 1px border-right set on the `bottom-actions` slot of the Action Bar. Maybe we can clean that up separately.

Re: design - this really doesn't have a visual of its own, the Action Bar is doing all of the UI work here aside from the bottom border on the slot, which seems correct to me.

**Review by macandcheese — APPROVED** [review]
Looks good, seems to behave as expected. Only visual thing of note is the border-right on the `bottom-actions` slot in Action Bar. Maybe conditionally don't set that based on the Action Bar layout?

**Comment by driskull** [comment]
> Only visual thing of note is the border-right on the bottom-actions slot in Action Bar.

@macandcheese that might not be related to this PR. I'm not setting any borders on the action-bar. Can you create a new issue for some CSS changes?

**Comment by driskull** [comment]
Since the action-bar goes horizontal now, should we rename "botttom-actions" to "end-actions"? Might want to reevaluate the names

**Comment by macandcheese** [comment]
> Since the action-bar goes horizontal now, should we rename "botttom-actions" to "end-actions"? Might want to reevaluate the names

Ha, yep see above issue. Agreed. I'll open issues for the border and slot name change on Action Bar.

File: src/components/panel/panel.stories.ts [hunk 1]
```diff
@@ -162,3 +162,15 @@ export const darkModeRTL_TestOnly = (): string =>
   );
 
 darkModeRTL_TestOnly.parameters = { modes: modesDarkDefault };
+
+export const withActionBar_TestOnly = (): string => `<calcite-panel height-scale="s">
```
Comment (jcfranco) [inline src/components/panel/panel.stories.ts:166]: Can you use the `html` formatting helper so prettier does its thang?

File: src/components/panel/panel.tsx [hunk 2]
```diff
@@ -358,6 +371,14 @@ export class Panel
     ) : null;
   }
 
+  renderActionBar(): VNode {
+    return (
+      <div class={CSS.actionBarContainer} hidden={!this.hasActionBar} key="action-bar">
```
Comment (jcfranco) [inline src/components/panel/panel.tsx:376]: If this is always rendered, `key` isn't needed.

**Review by jcfranco — APPROVED** [review]
🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥
🎥💥💥💥💥🎥🎥💥💥💥🎥💥💥💥💥💥🎥💥💥💥🎥🎥💥💥🎥🎥💥🎥🎥🎥💥🎥
🎥💥🎥🎥💥🎥💥🎥🎥🎥🎥🎥🎥💥🎥🎥🎥🎥💥🎥🎥💥🎥🎥💥🎥💥💥🎥🎥💥🎥
🎥💥💥💥💥🎥💥🎥🎥🎥🎥🎥🎥💥🎥🎥🎥🎥💥🎥🎥💥🎥🎥💥🎥💥🎥💥🎥💥🎥
🎥💥🎥🎥💥🎥💥🎥🎥🎥🎥🎥🎥💥🎥🎥🎥🎥💥🎥🎥💥🎥🎥💥🎥💥🎥🎥💥💥🎥
🎥💥🎥🎥💥🎥🎥💥💥💥🎥🎥🎥💥🎥🎥🎥💥💥💥🎥🎥💥💥🎥🎥💥🎥🎥🎥💥🎥
🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥🎥

**Pull request merged**